pycode/memilio-epidata/memilio/epidata/getVaccinationData.py

```diff
@@ -1066,6 +1131,140 @@ def get_vaccination_data(read_data=dd.defaultDict['read_data'],
+            - To do so getPopulationData is used and age group specific date from the original source
+                is extrapolated on the new age groups on county level.
+
+    - Missing dates are imputed for all data frames ('fillDates' is not optional but always executed).
+    - A central moving average of N days is optional.
+
+    - Start and end dates can be provided to define the length of the returned data frames.
+
+    @param read_data [Currently not used] True or False. Defines if data is read from file or downloaded.
+        Here Data is always downloaded from the internet.
+    @param file_format File format which is used for writing the data. Default defined in defaultDict.
+    @param out_folder Folder where data is written to. Default defined in defaultDict.
+    @param start_date Date of first date in dataframe. Default defined in defaultDict.
+    @param end_date Date of last date in dataframe. Default defined in defaultDict.
+    @param moving_average Integers >=0. Applies an 'moving_average'-days moving average on all time series
+        to smooth out effects of irregular reporting. Default defined in defaultDict.
+    @param sanitize_data: Value in {0,1,2,3}; Default: 1. For many counties,
+        vaccination data is not correctly attributed to home locations of
+        vaccinated persons. If 'sanitize_data' is set to larger 0, this is
+        corrected.
+        0: No sanitizing applied.
+        1: Averaging ratios over federal states.
+        2: Averaging ratios over intermediate regions.
+        3: All counties with vaccination quotas of more than
+        'sanitizing_threshold' will be adjusted to the average of its
+        federal state and remaining vaccinations will be distributed to
+        closely connected neighboring regions using commuter mobility networks.
+        The sanitizing threshold will be defined by the age group-specific
+        average on the corresponding vaccination ratios on county and federal
+        state level.
+    @param impute_dates bool True or False. Defines if values for dates without new information are imputed. Default defined in defaultDict.
+    @param to_dataset bool True or False. Whether to return the dataframe as an object instead of json file.
+        If True - returns objects with dataframes
+        If False - write dataframes into files
+        Default defined in defaultDict.
+
+    @return None
+    """
+    raw_df = fetch_vaccination_data(
+        read_data=read_data,
+        out_folder=out_folder,
+        **kwargs
+    )
+    process_df = process_vaccination_data(
+        df_data=raw_df,
+        start_date=start_date,
+        end_date=end_date,
+        file_format=file_format,
+        out_folder=out_folder,
+        moving_average=moving_average,
+        sanitize_data=sanitize_data
+    )
+    silver_datasets = write_vaccination_data(dict_data=process_df,
```

Comment:
what happens to the datasets? The main function does not have a return.

Reply:
I return these datasets because I use the write_function output in my pipeline and never the get_..._data function. So, there is no need for the main to return so to speak here. But, I should probably re-write a little more so we don't to these unnecessary returns

Reply:
Now, I have adjusted the return to be conditional